src/_pytest/python.py

```diff
@@ -800,7 +800,7 @@ def collect(self):
                     "%r generated tests with non-unique name %r" % (self, name)
                 )
             seen[name] = True
-            values.append(self.Function(name, self, args=args, callobj=call))
+            values.append(Function(name, self, args=args, callobj=call))
```

Comment:
this is a breaking change, not doing it is why i had massive trouble with the warnings